testing/tests/physics.lua

```diff
@@ -0,0 +1,685 @@
+  test:assertEquals(0, math.floor(shape1:getRadius()), 'check radius')
+  test:assertEquals('polygon', shape1:getType(), 'check rectangle type')
+  test:assertEquals(0, shape1:getBody():getX(), 'check body link')
+  test:assertEquals(1, shape1:getCategory(), 'check def category')
+  shape1:setCategory(3, 5, 6)
+  local categories = {shape1:getCategory()}
+  test:assertEquals(14, categories[1] + categories[2] + categories[3], 'check set category')
+  test:assertEquals(false, shape1:isSensor(), 'check sensor def')
+  shape1:setSensor(true)
+  test:assertEquals(true, shape1:isSensor(), 'check set sensor')
+  shape1:setSensor(false)
+  test:assertEquals(false, shape1:isDestroyed(), 'check not destroyed')
+  test:assertEquals(nil, shape1:getUserData(), 'check no user data')
+  shape1:setUserData({ test = 14 })
+  test:assertEquals(14, shape1:getUserData().test, 'check user data set')
+  -- check bounding box
+  -- polygons have an additional skin radius to help with collisions
+  -- so this wont be 0, 0, 10, 10 as you'd think but has an additional 0.3 padding
+  local topLeftX, topLeftY, bottomRightX, bottomRightY = shape1:computeAABB(0, 0, 0, 1)
+  local tlx, tly, brx, bry = shape1:getBoundingBox(1)
+  test:assertEquals(topLeftX, tlx, 'check bbox methods match tlx')
+  test:assertEquals(topLeftY, tly, 'check bbox methods match tly')
+  test:assertEquals(bottomRightX, brx, 'check bbox methods match brx')
+  test:assertEquals(bottomRightY, bry, 'check bbox methods match bry')
+  test:assertEquals(topLeftX, topLeftY, 'check bbox tl 1')
+  test:assertEquals(-3, math.floor(topLeftY*10), 'check bbox tl 2')
+  test:assertEquals(bottomRightX, bottomRightY, 'check bbox br 1')
+  test:assertEquals(10, math.floor(bottomRightX), 'check bbox br 2')
+  -- check physics props
+  test:assertEquals(1, shape1:getDensity(), 'check def density')
+  shape1:setDensity(5)
+  test:assertEquals(5, shape1:getDensity(), 'check set density')
+  local x, y, mass, inertia = shape1:getMassData()
+  test:assertEquals(5, math.floor(x), 'check shape mass pos x')
+  test:assertEquals(5, math.floor(y), 'check shape mass pos y')
+  test:assertEquals(5, math.floor(mass*10), 'check mass at 1 density')
+  test:assertEquals(0, math.floor(inertia*10), 'check intertia at 1 density')
+  x, y, mass, inertia = shape1:computeMass(1000)
+  test:assertEquals(111, math.floor(mass), 'check mass at 1000 density')
+  test:assertEquals(7407, math.floor(inertia), 'check intertia at 1000 density')
+  test:assertEquals(2, math.floor(shape1:getFriction()*10), 'check def friction')
+  shape1:setFriction(1)
+  test:assertEquals(1, shape1:getFriction(), 'check set friction')
+  test:assertEquals(0, shape1:getRestitution(), 'check def restitution')
+  shape1:setRestitution(0.5)
+  test:assertEquals(5, math.floor(shape1:getRestitution()*10), 'check set restitution')
+  -- check points
+  local shape2 = love.physics.newRectangleShape(body1, 5, 5, 10, 10)
```

Comment:
Should this be reusing the dynamic body, considering it's testing a world point below?